### PR TITLE
feat(chunking): add overlap

### PR DESCRIPTION
**Pull request by scanny**
Had to fix a few bugs first, which accounts for most of the code changes. Reads best commit-by-commit.

### CODE REVIEW

File: unstructured/chunking/title.py [hunk 1]
```diff
@@ -287,6 +310,7 @@ def wrapper(*args: _P.args, **kwargs: _P.kwargs) -> List[Element]:
                     combine_text_under_n_chars=params.get("combine_text_under_n_chars", 500),
                     new_after_n_chars=params.get("new_after_n_chars", 500),
                     max_characters=params.get("max_characters", 500),
+                    overlap=params.get("overlap", 0),
```
Comment (amanda103) [inline unstructured/chunking/title.py:313]: Just trying to think about the way this behaves with the other inputs:
1. what happens if overalp == max characters?
2. overlap > max characters - should error before going into chunking

Any other things we should be handling?

Comment (scanny) [reply unstructured/chunking/title.py:313]: Yes, that's a good point. I can add a validation step for that.

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,10 +2,15 @@
 
 ### Enhancements
 
+* **Add overlap option for split chunks.** Users can supply an `overlap: int` value on a partition function to start continuation chunks the specified number of characters "early", such that the last n characters of the prior chunk also appear as the first n characters of the following chunk, to improve semantic context.
+
 ### Features
 
 ### Fixes
 
+* **Fix out-of-order sequencing of split chunks.
```
Comment (cragwolfe) [inline CHANGELOG.md:11]:
```suggestion
* **Fix out-of-order sequencing of split chunks.**
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,10 +2,15 @@
...
+* **Fix breaking chunks on regex-metadata matches.
```
Comment (cragwolfe) [inline CHANGELOG.md:12]:
```suggestion
* **Fix breaking chunks on regex-metadata matches.**
```
Please say more. what does this mean, why does it matter?

File: CHANGELOG.md (hunk 2 above)
Comment (scanny) [reply CHANGELOG.md]: k, fixed

File: CHANGELOG.md (hunk 3 above)
Comment (scanny) [reply CHANGELOG.md]: k, fixed

**Comment by adanielyan** [comment]
Any idea when this will be merged to main?

**Comment by adanielyan** [comment]
Appreciate a quick response, @scanny!

My use case is related to populating a vector database with embeddings extracted from PDF files.

**Pull request closed without merging**